ambari-server/src/main/resources/stack-hooks/before-ANY/scripts/params.py

```diff
@@ -109,12 +109,22 @@ def is_secure_port(port):
 # which would cause a lot of problems when writing out hadoop-env.sh; instead
 # force the use of "current" in the hook
 hdfs_user_nofile_limit = default("/configurations/hadoop-env/hdfs_user_nofile_limit", "128000")
-hadoop_home = stack_select.get_hadoop_dir("home")
+
+mpack_name = config_helper.get_mpack_name(config)
+mpack_instance_name = config_helper.get_mpack_instance_name(config)
+module_name = config_helper.get_module_name(config)
+component_type = config_helper.get_component_type(config)
+component_instance_name = config_helper.get_component_instance_name(config)
+
+
+hadoop_home = get_component_home_path(mpack_name=mpack_name, instance_name=mpack_instance_name, module_name=module_name,
```

Comment:
Shouldn't this be hardcoded to HADOOP_CLIENTS/HADOOP_CLIENT as when starting for example HBASE this will be set incorrectly?

Reply:
Basically if the mpack = EDW and module = HBASE, you should be resolving hadoop_home to 

/usr/hwx/instances/edw/<mpack-instance-name>/hadoop_client (i.e. the hadoop_client in edw instance)

Reply:
But what if mpack doesn't have client module? Maybe we should use service or service group dependencies to define those?

Reply:
For the demo we need only configuration files that are populated in the hooks, we can just copy this logic into the hdfs script of hdpcore itself, to avoid modifying the hooks